widget/table.go

```diff
@@ -0,0 +1,328 @@
+		t.dividers[divs].Show()
+		divs++
+	}
+
+	i = 0
+	for y := theme.Padding() + t.scroll.Offset.Y - (t.scroll.Offset.Y % (t.cellSize.Height + tableDividerThickness)) - tableDividerThickness; y < t.scroll.Offset.Y+t.t.size.Height && i < rows-1; y += t.cellSize.Height + tableDividerThickness {
+		if y <= theme.Padding()+t.scroll.Offset.Y {
+			continue
+		}
+		i++
+
+		t.dividers[divs].Move(fyne.NewPos(theme.Padding(), y-t.scroll.Offset.Y))
+		t.dividers[divs].Resize(fyne.NewSize(t.t.size.Width-theme.Padding(), tableDividerThickness))
+		t.dividers[divs].Show()
+		divs++
+	}
+
+	for i := divs; i < len(t.dividers); i++ {
+		t.dividers[divs].Hide()
+	}
+}
+
+func (t *tableRenderer) Layout(s fyne.Size) {
+	t.moveIndicators()
+
+	t.scroll.Move(fyne.NewPos(theme.Padding(), theme.Padding()))
+	t.scroll.Resize(s.Subtract(fyne.NewSize(theme.Padding(), theme.Padding())))
+}
+
+func (t *tableRenderer) MinSize() fyne.Size {
+	return t.cellSize
+}
+
+func (t *tableRenderer) Refresh() {
+	template := t.t.NewCell()
+	t.cellSize = template.MinSize().Add(fyne.NewSize(theme.Padding()*2, theme.Padding()*2))
+	t.moveIndicators()
+
+	t.colMarker.FillColor = theme.PrimaryColor()
+	t.colMarker.Refresh()
+	t.rowMarker.FillColor = theme.PrimaryColor()
+	t.rowMarker.Refresh()
+
+	for _, div := range t.dividers {
+		div.(*canvas.Rectangle).FillColor = theme.ShadowColor()
+		div.Refresh()
+	}
+	t.t.cells.Refresh()
+}
+
+func (t *tableRenderer) BackgroundColor() color.Color {
+	return theme.BackgroundColor()
+}
+
+func (t *tableRenderer) Objects() []fyne.CanvasObject {
+	return t.objects
+}
+
+func (t *tableRenderer) Destroy() {
+}
+
+type tableCells struct {
+	BaseWidget
+	t        *Table
+	cellSize fyne.Size
+}
+
+func newTableCells(t *Table, s fyne.Size) *tableCells {
+	c := &tableCells{t: t, cellSize: s}
+	c.ExtendBaseWidget(c)
+	return c
+}
+
+func (c *tableCells) CreateRenderer() fyne.WidgetRenderer {
+	return &tableCellsRenderer{cells: c, pool: &syncPool{}, objects: []fyne.CanvasObject{}, visible: make(map[cellID]fyne.CanvasObject)}
+}
+
+func (c *tableCells) Tapped(e *fyne.PointEvent) {
+	if e.Position.X < 0 || e.Position.X >= c.Size().Width || e.Position.Y < 0 || e.Position.Y >= c.Size().Height {
+		c.t.SelectedColumn = -1
+		c.t.SelectedRow = -1
+		c.t.Refresh()
+		return
+	}
+
+	c.t.SelectedColumn = e.Position.X / (c.cellSize.Width + tableDividerThickness)
+	c.t.SelectedRow = e.Position.Y / (c.cellSize.Height + tableDividerThickness)
+
+	if c.t.OnCellSelected != nil {
+		c.t.OnCellSelected(c.t.SelectedRow, c.t.SelectedColumn)
+	}
+
+	if c.t.moveCallback != nil {
+		c.t.moveCallback()
+	}
+}
+
+type cellID struct {
+	row, col int
+}
+
+type tableCellsRenderer struct {
+	cells   *tableCells
+	pool    pool
+	objects []fyne.CanvasObject
+	visible map[cellID]fyne.CanvasObject
+}
+
+func (r *tableCellsRenderer) Layout(_ fyne.Size) {
+	// we deal with cached objects so just refresh instead
+}
```

Comment:
nit: could the layout code be moved here, and have Refresh call Layout?

Reply:
That would be slower as we would have to iterate all the cells twice.